appveyor.yml

```diff
@@ -31,18 +32,32 @@ before_build:
     - cd _build
     - perl ..\Configure %TARGET% %SHARED%
     - cd ..
+    - ps: >-
+        if (-not $env:APPVEYOR_PULL_REQUEST_NUMBER -or (&git log -2 | Select-String "\[extended tests\]") ) {
+            $env:EXTENDED_TESTS="yes"
+        }
 
 build_script:
-    - cd _build
-    - nmake
-    - cd ..
+    - ps: >-
+        If ($env:Configuration -Match "shared" -or $env:EXTENDED_TESTS) {
+            cd _build
+            &nmake
```

Comment:
It was desperate attempt to solve the problem with exception mentioned above. `&` so to say adds a scope level... In the context it might be worth keeping in mind that powershell is almost, but not quite, entirely unlike the kind of shell you'd imagine when you hear "shell" in its name.